core/src/test/java/org/testcontainers/containers/JibTest.java

```diff
@@ -0,0 +1,61 @@
+package org.testcontainers.containers;
+
+import com.google.cloud.tools.jib.api.CacheDirectoryCreationException;
+import com.google.cloud.tools.jib.api.Containerizer;
+import com.google.cloud.tools.jib.api.DockerDaemonImage;
+import com.google.cloud.tools.jib.api.InvalidImageReferenceException;
+import com.google.cloud.tools.jib.api.Jib;
+import com.google.cloud.tools.jib.api.JibContainer;
+import com.google.cloud.tools.jib.api.RegistryException;
+import org.junit.Test;
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.containers.startupcheck.OneShotStartupCheckStrategy;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class JibTest {
+
+    @Test
+    public void testJibFromWithString()
+        throws CacheDirectoryCreationException, IOException, ExecutionException, InterruptedException, RegistryException, InvalidImageReferenceException {
+        String image = "busybox:1.35";
+
+        JibContainer jibContainer = Jib
+            .from(image)
+            .setEntrypoint("echo", "Hello World")
+            .containerize(Containerizer.to(DockerDaemonImage.named("jib-hello-world")));
+
+        try (
+            GenericContainer<?> busybox = new GenericContainer<>(jibContainer.getTargetImage().toString())
```

Comment:
I think we would really benefit from something like `new GenericContainer<>(new JibImage(jib -> jib.from(image).setEntrypoint(...)))`, where `class JibImage implements Future<String>`. Otherwise, it doesn't feel actually integrated. WDYT?

Reply:
It would be great if we can set the docker client to the JibContainer and use tc abstractions there. I agree with this approach indeed I think I overthought this approach.

Reply:
Given we can set dockerClient in `Jib.from` and `Containerizer.to` as part of `containerize`, constructor `JibImage(Consumer<JibContainerBuilder> jibContainer)` was not added.

Reply:
This is done in the last commit. Description was also updated